### DISCUSSION TITLE
Workflow and stuff

### DISCUSSION THREAD

**Discussion by runkaiz** [Serene-Arc/bulk-downloader-for-reddit#371] [Q&A] [answered]
Sorry if this seems like a dumb question, I was just wondering about how should my workflow be if I were to store reddit with this tool. 

python -m bdfr archive x:\ --subreddit SuddenlyGay

thats an example of my archive script

python -m bdfr download x:\ --subreddit SuddenlyGay --make-hard-links

thats my download script.

So say that I want to download a subreddit, and without limits so around 1000 posts. Do I just run those two commands after one another and I would have the files and also the metadata like comments? Is there something else i can do to better organize it or does that pretty much sums up a basic workflow? Thanks for the help

**Comment by aliparlakci** [comment, marked as answer]
I don't understand what you exactly mean by "organize better" but these commands should be enough. Though, I strongly suggest you to read the [Options section](https://github.com/aliparlakci/bulk-downloader-for-reddit#options) in the README.

**Reply by runkaiz** [reply]
I guess what I wanted to say is that, I am pretty used to Sonarr and Radarr or similar automation softwares. I made some modifications to the flags but I was thinking if main idea of the bulk downloader is to use both archive and download together to get basically everything you need to reconstruct a Reddit (without css and stuff of course). Now that you clarified that I am confident about it so thanks! Also what if someone posts between my download and archive? I suppose if I sort by hot that shouldn't matter but what if a new post appears just while I was downloading, then if I archive the commands and stuff I would have one less json for the oldest post it saved towards the 1000 mark and an extra json for a content I didn't download yet. I haven't figure out how to run two instances of the bulk downloader yet, it refuse to detect my new directory for the second log file, should I manually make an empty log file for that?

**Reply by aliparlakci** [reply]
> it refuse to detect my new directory for the second log file

It looks like a bug. It would be great if you could open a bug report issue for this.

> Also what if someone posts between my download and archive?

I think we can modify the download process such that bdfr archives a submission as soon as it downloads it. I will open a feature request for it.
